### ISSUE TITLE
Vertex constructed with NAN in position

### ISSUE THREAD

**Issue by blackcathj** [sPHENIX-Collaboration/coresoftware#155]
As reported in https://github.com/sPHENIX-Collaboration/coresoftware/pull/148, vertex with NAN was submitted to loaded to the global vertex map, which crashed jet finder. A fix was implemented in #148 , vertex with NAN in z is rejected from filling into the global vertex map. Nevertheless, it does not solve the problem of how a NAN-vertex get produced in the first place. 

From the last simulation meeting, we decided to raise this issue again to keep track of this TODO item after ALD charge response.

**Comment by mccumbermike** [comment]
Can you give some detail on how to replicate? What kind of events should I toss?

**Comment by blackcathj** [comment]
Hi @mccumbermike 

Took a while to dig out the setup showing this problem, as it is already solved in recent builds (probably with the Eigen fix).

The problem concerns the builds before we fixed the tracking bug around May 16 2016. One example build that still kept on RCF is play.2 as built on May 8th. 

The event demonstrated this problem is Pythia8 jetty events with q^hat cut of 25GeV, which should have many tracks to reconstruct a good silicon vertex. The G4 simulation was analyzed with the default Fun4All macro. However, for a fraction of events, a NAN vertex is reconstructed. Here is a DST dump for such a event: 
```
[jinhuang@rcas2076 macros]$ root
root [0]   gSystem->Load("libg4eval.so");
root [1]   gSystem->Load("libg4vertex.so");
Warning in <TClassTable::Add>: class PHG4ScintillatorSlat already in TClassTable
root [2] TFile *_file0 = TFile::Open("G4sPHENIXCells.root")
root [3] T->Scan("DST.SVTX.SvtxVertexMap.get(0).identify():DST.BBC.BbcVertexMap.get(0).identify():DST.GLOBAL.GlobalVertexMap.get(0).identify()")
************************************************
*    Row   * DST.SVTX. * DST.BBC.B * DST.GLOBA *
************************************************
---SvtxVertex_v1--------------------
vertexid: 0
 t0 = 0
 (x,y,z) =  (0, 0, nan) cm  <---- invalid vertex from SVX
 chisq = 0,  ndof = 0
         ( 0 , 0 , 0 )
  err  = ( 0 , 0 , 0 )
         ( 0 , 0 , 0 )
 list of tracks ids: 
-----------------------------------------------
---BbcVertex_v1--------------------------------
vertexid: 0
 t = 0.000904271 +/- 0.002
 z =  -6.30308 +/- 0.6
-----------------------------------------------
---GlobalVertex_v1-----------------------
vertexid: 0
 t = 0
 (x,y,z) =  (0, 0, nan) cm  <---- propagated to the global vertex before fix #148
 chisq = 0,  ndof = 0
         ( 0 , 0 , 0 )
  err  = ( 0 , 0 , 0 )
         ( 0 , 0 , 0 )
 list of vtx ids: 
  2 => 0
```

In the most recent nightly build, this problem is gone. The same event is dumpped with succsufully reconstructed vertex in both BBC and SVX. The matched vertex is output to the ```GlobalVertex```. 
```
---SvtxVertex_v1--------------------
vertexid: 0
 t0 = 0
 (x,y,z) =  (0.00350489, 0.0221577, -6.38096) cm
 chisq = 0,  ndof = 0
         ( 0 , 0 , 0 )
  err  = ( 0 , 0 , 0 )
         ( 0 , 0 , 0 )
 list of tracks ids: 0 1 2 3 4 5 6 7 
-----------------------------------------------
---BbcVertex_v1--------------------------------
vertexid: 0
 t = 0.000904271 +/- 0.002
 z =  -6.30308 +/- 0.6
 -----------------------------------------------
---GlobalVertex_v1-----------------------
vertexid: 0
 t = 0.000904271
 (x,y,z) =  (0.00350489, 0.0221577, -6.38096) cm
 chisq = 0,  ndof = 0
         ( 0 , 0 , 0 )
  err  = ( 0 , 0 , 0 )
         ( 0 , 0 , 0 )
 list of vtx ids: 
  1 => 0
  2 => 0
-----------------------------------------------

```
You are welcome to close this issue.

**Comment by mccumbermike** [comment]
I noticed that the tracking module was sometimes letting through some temp vertexes without tracks, as these were. That problem was patched in the last merge, so we should be done with this problem for good. Closing...
